### PR TITLE
Ci fixes

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

CI is failing in various ways because of some trivial problems…

- Fix linting issues with README.md
- Bump all deps and prevent composer installation problems by denying the `php-http/discovery` composer plugin
- Fix deprecations in PHPUnit and `php-http/discovery`

### CODE REVIEW

**Comment by gsteel** [comment]
@Ocramius any ideas on improving MSI here? Most of the mutants are escaping via calls to `Shell\Execute` and it's not like that can be mocked 🤷‍♂️

File: infection.json.dist [hunk 1]
```diff
@@ -12,6 +12,6 @@
     "mutators": {
         "@default": true
     },
-    "minMsi": 98,
-    "minCoveredMsi": 100
+    "minMsi": 97.4,
+    "minCoveredMsi": 98
```
Comment (Ocramius) [inline infection.json.dist:16]: This dipping below 100% worries me a bit :D

Comment (Ocramius) [reply infection.json.dist:16]: > https://github.com/laminas/automatic-releases/pull/247#issuecomment-1717298203

I'd say that we can leave it be then - sad, but understandable.

**Review by Ocramius — APPROVED** [review]
Thanks @gsteel!

**Pull request merged**